File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/ServiceContracts/Northwind.TerritoryRow.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class TerritoryRow
    {
        public Int32? ID { get; set; }
        public String TerritoryID { get; set; }
        public String TerritoryDescription { get; set; }
        public Int32? RegionID { get; set; }
        public String RegionDescription { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string ID = "ID";
            [InlineConstant] public const string TerritoryID = "TerritoryID";
            [InlineConstant] public const string TerritoryDescription = "TerritoryDescription";
            [InlineConstant] public const string RegionID = "RegionID";
            [InlineConstant] public const string RegionDescription = "RegionDescription";
        }
    }
    
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.TerritoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface TerritoryRow {
    ID?: number;
    TerritoryID?: string;
    TerritoryDescription?: string;
    RegionID?: number;
    RegionDescription?: string;
}

export abstract class TerritoryRow {
    static readonly idProperty = 'ID';
    static readonly nameProperty = 'TerritoryID';
    static readonly localTextPrefix = 'Northwind.Territory';
    static readonly lookupKey = 'Northwind.Territory';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TerritoryRow>('Northwind.Territory') }
    static async getLookupAsync() { return getLookupAsync<TerritoryRow>('Northwind.Territory') }

    static readonly deletePermission = 'Northwind:General';
    static readonly insertPermission = 'Northwind:General';
    static readonly readPermission = 'Northwind:General';
    static readonly updatePermission = 'Northwind:General';

    static readonly Fields = fieldsProxy<TerritoryRow>();
}